src/linear_samplers/block_row_rand_replace.jl

```diff
@@ -0,0 +1,47 @@
+# This code was Written by Nathaniel Pritchard
+"""
+    LinSysBlkRowReplace <: LinSysBlkRowSampler 
+
+A mutable structure with fields to handle randomly subset block sampling with replacement, also known as
+vanilla block randomized kaczmarz. 
+
+# Fields
+- `block_size::Int64` - Specifies the size of each block.
+- `block::Vector{Int64}` - The list of all the rows in each block.
```

Comment:
Given that entries are added on the fly, is this just for memory considerations?

You also need to fix the `-` to `,` to keep it consistent with other docstrings

Reply:
This is preventing a new vector of indices being allocated at each function call.